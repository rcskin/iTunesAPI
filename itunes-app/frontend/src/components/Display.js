import React from "react";
import Logo from "../ituneslogo.png";

const Display = ({ searchItems, searchCategory, handleAddFavorite }) => {
  return (
    <div>
      {/* Display the list of searched items */}
      {searchItems && searchItems.length > 0 && (
        <div className="display-container">
          <h2>Search Results:</h2>
          {/* create an unordered list */}
          <ul className="display-items">
            {/* map over results array from API and render a list item for each 'result' */}
            {searchItems.map((result) => (
              // set the key prop of the list item to the unique trackId
              <li key={result.trackId}>
                <div className="display-item-details">
                  <div className="display-artwork">
                    {result.artworkUrl100 ? (
                      <img src={result.artworkUrl100} alt={result.trackName} />
                    ) : (
                      //if artwork is not available, display the iTunes logo
                      <img src={Logo} alt="itunes logo"></img>
                    )}
                  </div>
                  <div className="display-title">
                    <h3>
                      {searchCategory === "tvShow"
                        ? result.collectionName
                        : result.trackName || result.collectionName}
                    </h3>
                    <h4>
                      <i>{result.artistName}</i>
                    </h4>
                  </div>
                </div>
                {/* create button to add a result as a 'favorite' and call handleAdd function when clicked */}
                <button onClick={() => handleAddFavorite(result)}>
                  Add Favorite
                </button>
              </li>
            ))}
          </ul>
        </div>
      )}
    </div>
  );
};

export default Display;


