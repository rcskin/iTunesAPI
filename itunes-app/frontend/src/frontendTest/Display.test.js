import React from "react";
import renderer from "react-test-renderer";
import Display from "../components/Display";

test("Display component renders correctly", () => {
  // Define variables and functions needed for the test
  const searchItems = [];
  const searchCategory= "music";
  const handleAddFavorite = jest.fn();

  // Render the component and create a snapshot of the rendered tree
  const tree = renderer
    .create(
      <Display
        searchItems={searchItems}
        searchCategory={searchCategory}
        handleAddFavorite={handleAddFavorite}
      />
    )
    .toJSON();
  
  // Compare the rendered tree with the saved snapshot
  expect(tree).toMatchSnapshot();
});

/*Note to reviewer: used task notes and my previous L2T17 to help remember how to do a snapshot 
test. I had to do a lot of trial and error, but then realised I needed to define my props and 
use a mock function as described on jest literature.
Sources:
L2T17 Task notes
https://legacy.reactjs.org/docs/test-renderer.html
https://jestjs.io/docs/getting-started
https://jestjs.io/docs/mock-functions
*/
