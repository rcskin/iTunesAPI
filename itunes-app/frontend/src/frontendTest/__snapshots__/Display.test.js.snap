// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display component renders correctly 1`] = `<div />`;
